storage/storage-mq/src/api/amqp/amqpConsumer.ts

```diff
@@ -0,0 +1,71 @@
+import * as AMQP from 'amqplib'
+
+export default class AmqpConsumer {
+  private connection?: AMQP.Connection
+
+  public async init (amqpUrl: string, retries: number, msBackoff: number): Promise<void> {
+    for (let i = 0; i <= retries; i++) {
+      try {
+        this.connection = await this.connect(amqpUrl)
+        return
+      } catch (error) {
+        console.error(`Error initializing the AMQP Client (${i}/${retries}):
+        ${error}. Retrying in ${msBackoff}...`)
+        await this.sleep(msBackoff)
+      }
+    }
+    return Promise.reject(new Error(`Could not connect to AMQP broker at ${amqpUrl}`))
+  }
+
+  private async connect (amqpUrl: string): Promise<AMQP.Connection> {
+    try {
+      const connection = await AMQP.connect(amqpUrl)
+      console.log(`Connection to amqp host at ${amqpUrl} successful`)
+      return connection
+    } catch (error) {
+      console.error(`Error connecting to amqp host at ${amqpUrl}: ${error}`)
```

Comment:
I would consider a `console.log` as a form of error handling. Therefore, why are two methods (console.log and throw error) to handle the error?